### PR TITLE
feat(api): Add support for HTTP override

### PR DESCRIPTION
**Pull request by marcoboi**
**Pull Request checklist**

- [X] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [X] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -791,6 +791,19 @@ serialized properly:
...
 
+### HTTP Override
+
+Spark Job Server offers HTTP override functionality. 
+Often reverse proxies and firewall implement access limitations to, for example, DELETE and PUT requests.
+HTTP override allows overcoming these limitations by wrapping, for example, a DELETE request into a POST request.
+
+Requesting the destruction of a context can be accomplished through HTTP override using the following syntax: 
+    
+    $ curl -X POST "localhost:8090/contexts/test_context?_method=DELETE"
```
Comment (noorul) [inline README.md]: will curl -X DELETE stop working after this?

File: README.md [hunk 2]
```diff
@@ -791,6 +791,19 @@ serialized properly:
 
 If we encounter a data type that is not supported, then the entire result will be serialized to a string.
 
+### HTTP Override
```
Comment (noorul) [inline README.md]: You need to generate TOC using DocToc. See README.md

File: README.md (hunk 1 above)
Comment (marcoboi) [inline README.md]: Of course not, the override will be available on top of the standard methods.

**Comment by marcoboi** [comment]
sorry, accidentally closed the PR when pushing the change.
Just opened another one: #883

**Pull request closed without merging**